contents/node/nodejs.md

```diff
@@ -0,0 +1,209 @@
+
+### dependencies and devDependencies
+The `dependencies` property specify dependencies needed in production while the `devDependencies` property specifies dependencies needed in development.
+
+```json
+{
+  "dependencies": {
+    "express": "~3.0.1",
+    "sequelize":"latest",
+    "bluebird": "^3.4.7",
+    "angular":"latest",
+    "angular-ui-router": "~0.2.11",
+    "path":"latest",
+  },
+  "devDependencies": {
+    "eslint": "^4.16.0",
+    "eslint-config-airbnb-base": "^12.1.0",
+    "eslint-plugin-import": "^2.8.0"
+  }
+}
+```
+The `dependencies` property maps to an object that has the name and version range for each dependency. It is common to find carets (`^`) and tildes (`~`) in the version range. To understand what they are for, checkout [NodeSource's article on semvers](https://nodesource.com/blog/semver-a-primer/).
+
+You can find a wide range of useful packages on [npmjs](https://www.npmjs.com/). See the [module](# Node module system) section for how to import packages/modules.
+
+### scripts
+
+The `scripts` property is a useful tool that can be used to perform repetitive tasks such as testing and building. It takes in an object. For each property, the key is the given name of the command and the corresponding value is the actual command that will be run.
+
+For example, inside your package.json, you might have
+```json
+{
+  "scripts": {
+      "build": "node app.js",
+      "lint": "eslint **/*.js",
+      "lint-fix": "eslint --fix **/*.js"
+  }
+}
+```
+Running `npm run build` in the command shell will execute `node app.js` and similarly `npm run lint-fix` will fix your linting errors in your JavaScript files if you have eslint installed. To find out more about how you can use npm scripts refer to [freeCodeCamp's Introduction to NPM Scripts](https://medium.freecodecamp.org/introduction-to-npm-scripts-1dbb2ae01633).
+
+### Others
+For more information about `package.json`, refer to [npm official documentation](https://docs.npmjs.com/files/package.json)
+
+# Node module system
+
+Let's move on to the module system which play a central rode in Node. This system allows you to include other Javascript files and thus makes it easy to reuse external libraries and organize your code into separate parts with limited responsibilities.
+
+Node comes bundled with useful [core modules](https://nodejs.org/api/) such as the `fs` (file system) module which includes classes, methods and events to deal with file I/O operations and the `http` module which helps Node to transfer data over HTTP.
+
+There are also useful non-core modules maintained by the community and external developers such as *Bluebird* for promises.
+
+Importing modules is simple - just use the `require()` function and provide the module identifier or the file path.
+```js
+const http = require('http'); // import a core module
+const Promise = require('bluebird'); // import a non-core module
```

Comment:
perhaps you could illustrate how to install these modules and also add them as dependencies. ie `$ npm install bluebird --save`

Reply:
Amended the last commit and added this in the dependencies section and a few lines up